### PR TITLE
Add CoNSeP dataset

### PR DESCRIPTION
**Pull request by anwai98**
This PR adds nuclei segmentation dataset for H&E stained histopathology images (this is the original dataset used in HoVerNet publication).

NOTE: This is not available at the original hosted link, however there is an open-source tiled version of this data on Kaggle.

Q: @constantinpape: The image and respective instance labels are available as tiles. Stitching the labels as it is gives tiling window issues to the nuclei instances at the crop intersections. How do I ensure stitching the crops without the artifacts?

### CODE REVIEW

**Comment by anwai98** [comment]
Hi @constantinpape,

I tried to make stitching work. I am at the point where I got the interface overlaps, ready to compute the overlaps (using `from nifty.ground_truth import overlap`) (I saw the stitching functionality at elf uses this - I gave this a shot). I have a feeling that I am doing something wrong in getting the ids for overlapping instances (which is why I do not get the edge ids out of the computed rag).

Once you have time, it would be nice if you could hint me what am I still missing!

File: torch_em/data/datasets/histopathology/consep.py
Comment (constantinpape) [inline torch_em/data/datasets/histopathology/consep.py:103]: This all looks correct to me. If the issue persists after fixing the problem with the ids we need to debug this with a breakpoint together.

File: torch_em/data/datasets/histopathology/consep.py
Comment (anwai98) [reply torch_em/data/datasets/histopathology/consep.py:151]: Thanks for the mention. This is updated in the latest commit!

**Comment by anwai98** [comment]
Hi @constantinpape,

Looks like it worked for most of the instances. There are two things I still spot:
- Seems like some labels are missing out after stitching
- Some of the labels (very few in number) are not stitched. It's mostly small labels, but some objects are quite large and they do have obvious faces which should be merged.

**Comment by constantinpape** [comment]
Hi @anwai98 ,
the issue should be fixed now. I will wait for you to check it before we merge it.

**Comment by anwai98** [comment]
Thanks @constantinpape! This looks great. All the tiles are merged now!

There's one mystery still unsolved: there are objects which disappear after we stitch the labels (very few, but there are a couple per stitched labels).

(However, I think it's super minor and the fix made by you already takes care of preparing the data good enough for our evaluation concerns)

**Comment by constantinpape** [comment]
Didi you check with the latest version of https://github.com/constantinpape/elf/pull/106 @anwai98 ? (I know merged it, so you can also use the elf master). I updated the label stitching functionality there as well, and I think it should fix the issue you describe. Let me know if this does not fix it.

**Comment by anwai98** [comment]
Oh yeah you are right. I wasn't on the latest version of the PR. Now with the latest version merged at `master`, everything looks as expected.
Thank you so much!

**Comment by constantinpape** [comment]
Ok! Anything else to take care of here from your end, or should we merge it?

**Comment by anwai98** [comment]
Everything looks good to merge from my side. Thanks! 😁

**Review by constantinpape — APPROVED** [review]

**Pull request merged**
